src/config/vars.rs

```diff
@@ -2,6 +2,18 @@ use std::collections::HashMap;
 
 use regex::{Captures, Regex};
 
+// Environment variable names can have any characters from the Portable Character Set other
+// than NUL.  However, for Vector's interpolation, we are closer to what a shell supports which
+// is solely of uppercase letters, digits, and the '_' (that is, the `[:word:]` regex class).
+// In addition to these characters, we allow `.` as this commonly appears in environment
+// variable names when they come from a Java properties file.
+//
+// https://pubs.opengroup.org/onlinepubs/000095399/basedefs/xbd_chap08.html
+pub const ENVIRONMENT_VARIABLE_INTERPOLATION_PATTERN: &str = r"(?x)
```

Comment:
Nit: we have this pattern for static regexes:

https://github.com/vectordotdev/vector/blob/58d7f3dfb0b57445db931604c6f72d93015da505/lib/codecs/src/gelf.rs#L46

Would it make sense to use here too rather than building the regex for each call of `interpolate()`?